### PR TITLE
Option refactoring

### PR DESCRIPTION
**Pull request by Minyewoo**

### CODE REVIEW

File: lib/src/core/option/option.dart [hunk 1]
```diff
@@ -1,28 +1,18 @@
...
-  ///   - [onNone] - returns any default of type [R].
-  R fold<R>({
-    required R Function(T v) onSome,
-    required R Function() onNone,
-  });
+sealed class Option<T> {}
+/// 
+/// [Option] that holds value.
+final class Some<T> implements Option<T> {
+  final T value;
```
Comment (a-givertzman) [inline lib/src/core/option/option.dart:7]: why value is public ?

File: test/unit/core/option/option_fold_test.dart [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
   group('Option fold', () {
     test('calls onSome function if it is instance of Some', () {
       for(int i=-50; i<=50; i++) {
         final Option<int> option = Some(i);
-        final foldResult = option.fold(
-          onSome: (v) => '$v',
-          onNone: () => 'None',
-        );
+        final foldResult = switch(option) {
+          Some<int>(:final value) => '$value',
```
Comment (a-givertzman) [inline test/unit/core/option/option_fold_test.dart:15]: remove int to string conversion, test for int,
batter if test will have such test dataset: 
```
testData = [
   int.MIN,
   int.MAX,
   double...
   bool...
]
```

File: test/unit/core/option/option_fold_test.dart [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
       }
     });
     test('calls onNone function if it is instance of None', () {
       const Option<int> option = None();
-      final foldResult = option.fold(
-          onSome: (v) => '$v',
-          onNone: () => 'None',
-        );
+      final foldResult = switch(option) {
+        Some<int>(:final value) => '$value',
```
Comment (a-givertzman) [inline test/unit/core/option/option_fold_test.dart:24]: remove int to string conversion, test for int

**Review by a-givertzman — COMMENTED** [review]
I suggested an example, please try to improve all tests for Option and Result

File: test/unit/core/option/option_fold_test.dart (hunk 2 above)
Comment (Minyewoo) [reply test/unit/core/option/option_fold_test.dart]: done

File: test/unit/core/option/option_fold_test.dart (hunk 3 above)
Comment (Minyewoo) [reply test/unit/core/option/option_fold_test.dart]: done

**Review by a-givertzman — APPROVED** [review]

**Pull request merged**